src/oc_erchef/apps/chef_objects/src/chef_user.erl

```diff
@@ -316,7 +316,16 @@ common_user_validation(EJ, User, Operation) ->
     end,
     % Ensure the user can't set this - we make use of it in some cases by setting it internally
     EJ1 = ej:delete({<<"id">>}, EJ),
-    {ok, EJ1}.
+    % Setting the email to lower case for uniqueness,
```

Comment:
Instead of doing this explicitly, could we let the database manage it by updating the index to be case-insensitive?

There's a risk there because it could cause a migration to fail if multiple case versions of the same email are already in the users table - but given that /is/ an error condition that we're not allowing, it's probably a better route.  We could pre-emptively identify that in upgrade preflight checks, and give the user instructions to mitigate it before proceeding.

Reply:
Yeah - this was one of the approaches we considered, but since this is required more internally than for our customers, we decided to not force data migration / validation on them.

Reply:
When we first made this change (unique email requirement, a few years ago)  not having a case-insensitive index in the first place was an oversight -- the intent was to not have any duplicate emails in the system. Enforcing data constraints at the application level is more likely to let edge cases through, which overall combines to make the DB a better place for this change. 

With all that said - practically speaking email isn't used in non-hosted solutions beyond org invites, password resets, and email verification; unique emails are not critical to the operation of a self-hosted CIS and we do want to limit the upgrade impact to customers.